assets/js/modules/tagmanager/setup.js

```diff
@@ -420,19 +417,16 @@ class TagmanagerSetup extends Component {
 							onEnhancedChange={ this.handleContainerChange }
 							outlined
 						>
-							{ containers.map( ( container ) => {
-								return (
-									<Option
-										key={ container.containerId /* Capitalization rule exception: `containerId` is a property of an API returned value. */ }
-										value={ container.publicId /* Capitalization rule exception: `publicId` is a property of an API returned value. */ }>
-										{
-											0 === container.publicId ? // Capitalization rule exception: `publicId` is a property of an API returned value.
-												__( 'Set up a new container', 'google-site-kit' ) :
-												container.publicId /* Capitalization rule exception: `publicId` is a property of an API returned value. */
-										}
-									</Option>
-								);
-							} ) }
+							{ containers.concat( {
+								name: __( 'Set up a new container', 'google-site-kit' ),
+								publicId: 0,
+							} ).map( ( { name, publicId }, i ) =>
+								<Option
+									key={ i }
+									value={ publicId /* Capitalization rule exception: `publicId` is a property of an API returned value. */ }
+									children={ name }
```

Comment:
Why do it this way and not like commonly done in React with proper child nesting `<Option ...>{ name }</Option>`?

Reply:
This is just a stylistic preference which seemed a bit cleaner at the time, particularly since the component is spread vertically. If it were `<Option>value</Option>` without other props I definitely would have written it that way. Happy to change this if you'd like.

Reply:
I think for consistency we should always pass children as proper child-like nodes. Will update it accordingly quickly.